src/surface/photosphere/fisheye_projection.py

```diff
@@ -0,0 +1,182 @@
+import math
+from dataclasses import dataclass
+
+import cv2
+import numpy as np
+
+
+@dataclass
+class FisheyeImage:
+    img: np.ndarray
+    img_num: int
+    left: int
+    top: int
+    diameter: int
+
+
+# The dimensions of the output image in width by height
+output_dimension = (2000, 1000)
+
+# The apeture of the fisheyes in radians
+apeture = 195 * math.pi / 180
+
+# The fisheye images with their information
+fisheye_images = [
+    FisheyeImage(
+        img=cv2.imread('src/surface/photosphere/fisheye1.jpg'),
+        img_num=0,
+        left=400,
+        top=28,
+        diameter=3052,
+    ),
+    FisheyeImage(
+        img=cv2.imread('src/surface/photosphere/fisheye2.jpg'),
+        img_num=1,
+        left=404,
+        top=-25,
+        diameter=3040,
+    ),
+]
+
+
+# Converts the coordinates in the projection to coordinates in the fisheye image
+# all coordinates are in unit coordinates (-1 to 1 in all dimensions with 0 being center)
+def projection_to_fisheye(x: float, y: float, img: int) -> list:
+    # Equirectangular to latitude/longitude
+    theta = math.pi * x + math.pi * img
+    phi = math.pi * y / 2
+
+    # Latitude/Longitude to 3D vector
+    px = math.cos(phi) * math.sin(theta)
+    py = math.cos(phi) * math.cos(theta)
+    pz = math.sin(phi)
+
+    # 3D vector to 2D fisheye
+    r = 2 * math.atan2(math.sqrt(px * px + pz * pz), py) / apeture
+    theta = math.atan2(pz, px)
+
+    x = r * math.cos(theta)
+    y = r * math.sin(theta)
```

Comment:
Maybe use different variable names here to indicate that these are fisheye coords